src/python/pants/backend/python/lint/black/rules.py

```diff
@@ -32,16 +32,21 @@
 class BlackSetup:
   """This abstraction is used to deduplicate the implementations for the `fmt` and `lint` rules,
   which only differ in whether or not to append `--check` to the Black CLI args."""
-  config_path: Optional[Path]
   resolved_requirements_pex: Pex
+  config_path: Optional[Path]
+  wrapped_target: FormattablePythonTarget
   merged_input_files: Digest
 
-  def generate_pex_arg_list(self, *, files: Tuple[str, ...], check_only: bool) -> Tuple[str, ...]:
+  def generate_args(
+    self, *, check_only: bool, passthrough_args: Optional[Sequence[str]] = None,
```

Comment:
I think this should take the files in a param, like before, but not take the passthrough args, as they don't change between targets, so they really are part of the setup.